articles/embedded.md

```diff
@@ -0,0 +1,309 @@
+2.0 stack for microcontrollers.
+
+*   [**ROS 2 library for OpenCR by ROBOTIS**](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2): 
+Tailored and optimized implementation of the ROS 2 publish/subscribe 
+and clock API for the Micro XRCE-DDS (formerly micro-RTPS) implementation 
+of the upcoming XRCE-DDS middleware standard running on an STM32F7.
+
+*   [**XEL Network by ROBOTIS**](https://xelnetwork.readthedocs.io):
+  Product which communicate with ROS 2 (DDS) through DDS-XRCE using Micro 
+  XRCE-DDS in the firmware of their CommXel board.
+  This CommXel board manages the rest of the boards conforming the XEL 
+  Network and interface them to a ROS 2 space.
+  The CommXel could use Ethernet or UART to communicate using DDS-XRCE.
+
+*   [**freeRTPS**](https://github.com/ros2/freertps): A free, portable, 
+minimalist implementation of the RTPS protocol for microcontrollers such as 
+the STM32F7 developed at the OSRF. FreeRTPS shall allow to run ROS 2 with 
+standard DDS as-is on stronger MCUs. This project has been discontinued in 2016.
+
+*   [**ros2_embedded_nuttx**](https://github.com/ros2/ros2_embedded_nuttx): 
+Early port (in 2014) of ROS 2 alpha for the STM32F4Discovery board and the 
+STM3240G eval board running the RTOS NuttX developed by Víctor Mayoral Vilches 
+and Esteve Fernandez at the OSRF.
+
+*   [**Renesas GR-ROSE**](http://gadget.renesas.com/ja/event/2018/pm_rose.html):
+  Renesas have integrated their GR-ROSE platform with ROS 2 using DDS-XRCE protocol.
+  They use Micro XRCE-DDS implementation on top of FreeRTOS.
+  A sample can be found in their forums 
+  [renesas forum ](https://japan.renesasrulz.com/gr_user_forum_japanese/f/gr-rose/5201/ros-2-micro-rtps).
+  They have integrated Micro XRCE-DDS middleware as part of their
+  [online web compiler](http://tool-cloud2.renesas.com/index.php) for the GR-ROSE platform.
+
+### ROS1-based approaches
+
+*   [**rosserial**](http://wiki.ros.org/rosserial): Well-known and widely used 
+in the ROS community.
+
+*   [**mROS**](https://github.com/tlk-emb/mROS/): A new 
+    work on bringing ROS1 concepts (including nodes and the 
+    ROS1 middleware) on stronger MCUs, cf.
+    *Hideki Takase, Tomoya Mori, Kazuyoshi Takagi and Naofumi Takagi: 
+    'Work-in-Progress: Design Concept of a Lightweight Runtime Environment 
+    for Robot Software Components onto Embedded Devices' in 
+    Proc. of ESWEEK, Torino, Italy, September 2018.*
+
+## Design Discussion
+
+The following figure may serve as a starting point for the design discussion.
+It depicts the major layers from the real-time operating system to the 
+application, in the style the ROS 2 standard stack.
+
+![micro-ROS](/img/embedded/features_with_dependencies.png)
+
+At the same time, the diagram illustrates the possible feature set of the 
+client library -- ideally in a modular fashion so that different profiles 
```

Comment:
This looks like a great starting point for talking about how to improve the structure of rcl to allow major features to be turned on and off at compile time.